### DISCUSSION TITLE
關於繁體中文翻譯/The question about how to translate MAS mod into traditional Chinese

### DISCUSSION THREAD

**Discussion by monikaanderic** [Monika-After-Story/MonikaModDev#9129] [QA - tech support + general] [answered]
大家好，我是一位繁體中文使用者，想要請問是否有人知道如何將莫妮卡模組的文字轉換成繁體中文。雖然網路上有一些簡體中文版本，但是我希望能夠使用繁體中文版本。為了得到更多人的回應與理解，我會同時使用中文與英文來描述我的問題。這是我第一次在github上發問，如果有任何錯誤，請接受我的歉意。
Hello, I am a traditional Chinese user, I want to translate MAS mod into traditional Chinese. Although there some Simplified Chinese translation on the Internet, I hope I can use traditional Chinese version. In order to get more answer and let you know what I mean, I will use traditional Chinese and English to describe my question. It is my first time to ask questions on github, if there are any mistake, please accept my apologies.

1.我知道需要將script-topics.rpy這類的rpy檔案用Ren'Py軟體打開，並把裡面的對話翻譯成中文，但我欠缺此方面的知識，不知道該怎麼操作。若有人能提供流程，我會很感謝。
I know that I should open rpy files (like"script-topics.rpy") by using Ren'Py, and translate the dialogues in files into traditional Chinese. However, I don't have any knowledge about this area and I have no idea about how to operate it. If anyone can provide operational process, I will be very grateful.

2.在閱讀script-topics.rpy時，我發現會出現這樣的文字:m 1eua "I'm glad nothing bad has happened to you while learning."我知道m代表Monika，但是我不懂1eua的意思。那是在控制Monika的表情嗎?
When reading script-topics.rpy, I find the sentences like "  m 1eua "I'm glad nothing bad has happened to you while learning."  "
I know that "m" means "Monika", but I don't understand what is  "1eua". Does it control Monika's emotions?

3.為甚麼rpy檔案在電腦上的分類是RPYC檔案?兩者有何不同?
Why does my coputer classify rpy files into "RPYC"file? What is the difference between them?

不管你使用甚麼語言來回應我，我都會非常感謝。祝大家有美好的一天。
No matter what language you use, I will be very grateful if you reply me. Have a nice day .

**Comment by Wingdinggaster656** [comment, marked as answer]
WARNING: I AM USING TRANSLATOR, SO MY CHINESE WORDS MAY NOT ACCURATE.
警告：我正在使用翻译软件，所以我的中文字符不一定精准。

For question 1:
You said you want to translate to Chinese, though you "don't know"? You don't know what? You don't know how to translate, or you don't know how to present them in game?
对于问题1：
你说你想要翻译成中文，但是你“不知道”？你不知道什么？你不知道怎么翻译，还是你不知道怎么把它们呈现在游戏里？

For question 2:
Yes it's expression things. Kind of complicated, and I don't think you need or you should change the expression, so I won't explain them.
对于问题2：
是的，是表达式这类东西。有点复杂，而且我不认为你需要或者你应该改变表达式，所以我不会解释它们。

For question 3:
Did you see Cocos engine output their game with `.cpp` files? You could consider `.rpy` as the source codes, consider `.rpyc` as the `.exe` file. In other words, `.rpyc` is the compiled `.rpy`. You should only let your `.rpyc` files appear in the game directory.
对于问题3：
你见过Cocos引擎用`.cpp`文件来输出它们的游戏吗？你可以将`.rpy`认为是源代码，将`.rpyc`认为是'.exe'文件。换言之，`.rpyc`是编译后的`.rpy`. 你应该只让`.rpyc`文件出现在游戏目录里。

**Reply by monikaanderic** [reply]
Thanks for your answer!
In fact, I don't know how to use Ren'Py.
I make a new project in Ren'Py, put "script-topics.rpy" into the project, and click the"script.rpy" button under "Edit File". However, it shows that my computer can not open it. It said that I should download other app to open it.
![2022-05-15 (10)](https://user-images.githubusercontent.com/105575736/168464943-19011d79-35a6-4697-8bac-2d7e2b2ff645.png)

**Reply by Wingdinggaster656** [reply]
You mean you are having trouble with compiling rpy files?
I have to warn you firstly that it would be pointless to separately put any of the MAS rpy files, because they don't run on their own (e.g. Topics file requires a lot of event handler functions). The only point of putting them in is to compile them.
Renpy program compiles their `/game` directory files every time the program started. 

This is not the most formal way to compile rpy files, but since you'll probably need to debug them frequently, I think you should just put your modified rpy files in your MAS's `/game` directory.  This will overwrite the old `script-topics.rpyc`.

**Reply by monikaanderic** [reply]
Thank you! I will try it again in correct way.

**Reply by Wingdinggaster656** [reply]
By the way, I personally suspect you haven't even read the documentation for Renpy.

You can find the Renpy documents here: https://www.renpy.org/doc/html/

I also found two Chinese document in that page: https://renpy.cn/doc/ and https://renpy.cn/doc-tw/
Really unsure about why two pages. I guess one of them is Simplified Chinese, and the other one is Traditional Chinese.

I don't know the difference between these two ways to write Chinese. It's hard to tell them apart. See for yourself which version you need.

(I don't know if these are two languages or two ways of writing the same language. I've heard it is "two ways of writing the same language". I hereby declare that I do not discriminate in any way.)

**Reply by monikaanderic** [reply]
Thank you so much! I didn't find that because it's my first day to use this website and Renpy. You really help me a lot!
